app/models/email_log.rb

```diff
@@ -14,6 +14,9 @@ class EmailLog < ActiveRecord::Base
     signup_after_approval
   }
 
+  # cf. https://www.iana.org/assignments/smtp-enhanced-status-codes/smtp-enhanced-status-codes.xhtml
+  SMTP_ERROR_CODE_REGEXP = Regexp.new(/\d\.\d\.\d|\d\d\d/).freeze
```

Comment:
That won't work for `X.7.13` for example. 

Something like this should work `\d\.\d\.\d+|\d{3}`

Reply:
@ZogStriP follow up here, ty for pointing this out :) https://github.com/discourse/discourse/pull/16019